.changelog/21251.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+core: Fix multiple incorrect type conversion for potential overflows
```

Comment:
Do we need to link the CVE here or does this security fix not have one

Reply:
Good question, this one would not need a link to a CVE